packages/k6-generator/src/index.js

```diff
@@ -0,0 +1,260 @@
+#!/usr/bin/env node
+
+const fs = require('fs');
+const replace = require('replace-in-file');
+const { StringPrompt } = require('enquirer');
+
+const templateFile = fs.readFileSync('./template.js');
+
+let [customImport, customMetrics, writeVuObj, uniqueObj, mainContent] = ['', '', '', '', ''];
+
+module.exports.run = async () => {
+  const methodsPath = await new StringPrompt({
+    initial: './methods/expressTemplate.json',
+    message: 'Confirm the path to the methods JSON file:',
+  }).run();
+
+  if (
+    !fs.existsSync(`${__dirname}${methodsPath.slice(methodsPath.indexOf('/'))}`) ||
+    methodsPath.slice(-5) !== '.json'
+  ) {
+    console.log(`${methodsPath} is not a valid path`);
+    process.exit();
+  }
+
+  const methods = require(`${__dirname}${methodsPath.slice(methodsPath.indexOf('/'))}`);
+
+  const newFile = await new StringPrompt({
+    initial: 'tests/k6/scripts/alpha.js',
+    message: 'Choose path and name for the results file',
+  }).run();
+
+  if (newFile.slice(-3) !== '.js') {
+    console.log(`${newFile} must end in '.js'`);
+    process.exit();
+  }
+
+  fs.writeFileSync(newFile, templateFile);
+
+  // DETERMINE IF UNIQUE DATA IS NEEDED
+  for (let i = 0; i < Object.keys(methods.routes).length; i++) {
+    let methodUrl = Object.keys(methods.routes)[i];
+
+    for (let j = 0; j < methods.routes[methodUrl].length; j++) {
+      const current = methods.routes[methodUrl][j];
+      const { path, uniquePayload } = current;
+
+      // GET FILE OF UNIQUE DATA, PARSE NAME OF OBJECT CONTAINED WITHIN
+      if (uniquePayload) {
+        const uniqueFile = await new StringPrompt({
+          initial: '../unique/user.json',
+          message: `${methodUrl} request to ${path} requires unique payload - provide a path to JSON file containing this data`,
+        }).run();
+
+        if (
+          !fs.existsSync(`${__dirname}${uniqueFile.slice(uniqueFile.indexOf('/'))}`) ||
+          uniqueFile.slice(-5) !== '.json'
+        ) {
+          console.log(`${uniqueFile} is not a valid path`);
+          process.exit();
+        }
+
+        customImport = `const SLEEP_DURATION = ${methods['sleep']}\nconst uniqueData = JSON.parse(open('${uniqueFile}'))`;
+        uniqueObj = `'${uniqueFile.slice(uniqueFile.lastIndexOf('/') + 1, -5)}'`;
+      }
+      if (!customImport) {
+        customImport = `const SLEEP_DURATION = ${methods['sleep']}`;
+      }
+    }
+  }
+
+  if (!uniqueObj) {
+    uniqueObj = '""';
```

Comment:
What is this for?

Reply:
This is because replace-in-file doesn't keep the first set of quotes - so when there was no unique object to define, I would end up with something like `uniqueObj = ;` which would give me an error